### PR TITLE
Implement Latex Renderer

### PR DESCRIPTION
**Pull request by laylaylo**
### Changes
- configuration of flutter_tex by [this guide](https://pub.dev/packages/flutter_tex) --> _android, iOS, web, pubspec_ files 
- applying it as **TexView()** to node view page, for theorems and proofs --> _node_details, proof_list_view_
- applying it as **TexView()** to graph page for theorem content --> _graph_node_card, graph_node_popup_
- fixing some weird characters by utf8.decode for theorem, proof, and also title: applied it anywhere related I saw

### Testing
manual 
everything seems fine but TexView() shows normal text content as empty, noticed on node/99 first proof. However, I plan to solve it by modifying DB to have everything according to some format (latex, latex etc.)

_note:_ I felt like scrolling becomes harder also on node/99 proofs, but not sure. I hope it's just me (my pc having hard times with all flutter and android studio)🤞🏻

### CODE REVIEW

**Comment by omerfaunal** [comment]
There are some internal conflicts here regarding files created by flutter. It must be fixed carefully. Let me try to fix it.

**Comment by laylaylo** [comment]
@omerfaunal Yes, as I see these conflicts are due to  Search Semantic UI PR #509, so I believe we need to take all part of both frontend and this branch on those files. I didn't solve them yet since I want to be sure about Semantic related part. We should keep all changes, right?

File: project/FrontEnd/collaborative_science_platform/lib/screens/graph_page/widgets/graph_node_popup.dart [hunk 1]
```diff
@@ -89,11 +91,14 @@ class _NodeDetailsPopupState extends State<NodeDetailsPopup> {
...
-              AnnotationText('Title: ${node.nodeTitle}'),
+              AnnotationText('Title: ${utf8.decode(node.nodeTitle.codeUnits)}'),
               Text(
                   'Contributors: ${node.contributors.map((user) => "${user.firstName} ${user.lastName} (${user.email})").join(", ")}'),
               Text('Publish Date: ${getDurationFromNow(node.publishDate!)}'),
-              AnnotationText('Theorem Content: ${node.theorem?.theoremContent ?? "No theorem"}'),
+              TeXView(
+                  renderingEngine: TeXViewRenderingEngine.katex(),
+                  child: TeXViewDocument(
+                      '<b>Theorem Content:</b> ${utf8.decode(node.theorem!.theoremContent.codeUnits) ?? "No theorem"}')),
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/screens/graph_page/widgets/graph_node_popup.dart:101]: Also this part got my attention. You are using "!" operator which is telling this value can't be null.
So you are claiming theorem can't be null. However it indeed can be. You need to do something like this.

node.theorem == null ? "No theorem" : node.theorem.theoremContent.codeUnits


I'll fix this along conflicts

**Review by omerfaunal — CHANGES_REQUESTED** [review]

File: project/FrontEnd/collaborative_science_platform/lib/screens/graph_page/widgets/graph_node_popup.dart (hunk 1 above)
Comment (omerfaunal) [reply project/FrontEnd/collaborative_science_platform/lib/screens/graph_page/widgets/graph_node_popup.dart]: I resolved this

**Review by omerfaunal — APPROVED** [review]

**Pull request merged**